### PR TITLE
Fix errors reported in DataDog

### PR DESCRIPTION
**Pull request by we-ai**
Updates in this PR:
- Fixes on recent errors in SMDB reported in DataDog, detailed in [#748](https://github.com/episphere/connect/issues/748)
- Clean-ups

### CODE REVIEW

File: siteManagerDashboard/participantWithdrawal.js [hunk 1]
```diff
@@ -68,25 +68,33 @@ const checkPreviousWithdrawalStatus = (participant) => {
...
-    if (participant[fieldMapping.refusalOptions][fieldMapping.refusedQualityOfLifeSurvey] === fieldMapping.yes ) template += `Refused QOL survey 3-mo (but willing to do other future surveys)​, `
-    if (participant[fieldMapping.refusalOptions][fieldMapping.refusedAllFutureQualityOfLifeSurveys] === fieldMapping.yes ) template += `Refused all future QOL surveys (but willing to do other future surveys)​, `
-    if (participant[fieldMapping.refusalOptions][fieldMapping.refusedFutureSurveys] === fieldMapping.yes ) template += `All future surveys (willing to do specimens)​, ` 
-    if (participant[fieldMapping.refusedAllFutureActivities] === fieldMapping.yes ) template += `All Future Study Activities, ` 
-    if (participant[fieldMapping.revokeHIPAA] === fieldMapping.yes ) template += `Revoke HIPAA Authorization, `
-    if (participant[fieldMapping.withdrawConsent] === fieldMapping.yes ) template += `Withdraw Consent​, `
-    if (participant[fieldMapping.destroyData] === fieldMapping.yes ) template += `Destroy Data​, `
-    if (participant[fieldMapping.participantDeceased] === fieldMapping.yes ) template += `Participant Deceased, ` 
+  let strArray = [];
```
Comment (JoeArmani) [inline siteManagerDashboard/participantWithdrawal.js:71]: Nice refactor/pattern!

File: siteManagerDashboard/participantDetailsHelpers.js [hunk 2]
```diff
@@ -790,7 +790,7 @@ export const saveResponses = (participant, changedOption, editedElement, concept
                 changedOption[currentConceptId] = newValueElement.value;
                 // if a changed field is a date of birth field then we need to update full date of birth  
                 if (fieldMapping.birthDay in changedOption || fieldMapping.birthMonth in changedOption || fieldMapping.birthYear in changedOption) {
-                    const day =  changedOption[fieldMapping.birthDay] || participant[fieldMapping.birthDay];
+                    const day =  changedOption[fieldMapping.birthDay] || participant[fieldMapping.birthDay] || '01'; // Not ideal here
```
Comment (JoeArmani) [inline siteManagerDashboard/participantDetailsHelpers.js:793]: What if we extract the fallback from fieldMapping.dateOfBirthComplete and use '01' as the third fallback? I didn't realize we were having missing value issues here.

**Review by JoeArmani — APPROVED** [review]
one small comment about birthday 'day' value

looks good

File: siteManagerDashboard/participantWithdrawal.js [hunk 3]
```diff
@@ -68,25 +68,33 @@ const checkPreviousWithdrawalStatus = (participant) => {
...
+  const refusalOptions = participant[fieldMapping.refusalOptions];
```
Comment (Gbarra9) [inline siteManagerDashboard/participantWithdrawal.js:72]: I like how you added the `refusalOptions` variable to make the conditionals below easier to read.

**Review by Gbarra9 — APPROVED** [review]

**Pull request merged**